### PR TITLE
Justine staminaglow

### PR DESCRIPTION
**Pull request by justineechen**
- isRestricted boolean for zones that are restricted , this allows the stamina bar to appear when they're in the zone
- isFirst boolean to be removed when alex calls it, this shows the glow animation that gives a visual cue to the user to introduce this feature of the game

### CODE REVIEW

File: Assets/Scripts/CameraZone.cs [hunk 1]
```diff
@@ -31,6 +33,18 @@ private void OnTriggerEnter(Collider other)
     {
         if (other.CompareTag(Constants.TAG_PLAYER))
         {
+            if (isFirst && isRestricted) //to be removed when alex finds where he calls it
+            {
+                StartCoroutine(UIManager.Instance.staminaBar.Glow());
+            }
+            if (isRestricted)
+            {
+                UIManager.Instance.staminaBar.FadeOut();
```
Comment (jurcaua) [inline Assets/Scripts/CameraZone.cs]: I think it would make more sense if "FadeIn" meant that it was visible right? Maybe the names should be switched.

File: Assets/Scripts/StaminaBar.cs [hunk 2]
```diff
@@ -70,4 +86,49 @@ void UpdateFillAmount(float newFill)
...
+        yield return null;
+    }
+
+    public IEnumerator Glow()
+    {
+        glowOutline.SetActive(true);
+        Animator animator = glowOutline.GetComponent<Animator>();
+        animator.SetBool("glow", true);
+        
+        yield return new WaitForSeconds (4);
```
Comment (jurcaua) [inline Assets/Scripts/StaminaBar.cs]: Should this `4` be a public variable as well?

**Review by jurcaua — CHANGES_REQUESTED** [review]
Gonna try it out manually now too to take a look!!! Very nice stuff though and just what I expected the logic to look like :)

**Comment by jurcaua** [comment]
I'd make the Glow part of the StaminaBar prefab, they're currently disconnected:
image

This would also help like save references to the greyOutlineImage and glowOutline in the code since the prefab will save them. 
If you need to move around stuff in the prefab to make it work (like making the glow a parent of everything or whatever you need to do, feel free!)

**Comment by jurcaua** [comment]
I think for the animation, you should just like make a single animation that'll do the four glows! 
Then you can simplify the animation tree to just `Idle` and `Glow`! The 4 states right now isn't too ideal cause it's a bit hard to follow: 
image

You may be able to simplify the Glow logic just to a single function too,
if you made a "trigger" in the animation controller (instead of a bool), and then you can do `.SetTrigger("Glow")`, with an animation transition that went from Idle to Glow using that trigger, and then just had a transition from Glow to Idle with no conditions (so it will always go back)

then you wont need a coroutine anymore cause you can just SetTrigger and itll finish on its own

File: Assets/Scripts/StaminaBar.cs [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
 using UnityEngine;
 using UnityEngine.UI;
 
 public class StaminaBar : MonoBehaviour
 {
     public Image staminaFillImage;
+    public Image greyOutlineImage;
+    public GameObject glowOutline;
+    private Image glowOutlineImage;
+    public float glowSeconds = 4f;
```
Comment (jurcaua) [inline Assets/Scripts/StaminaBar.cs:12]: This isn't being used anymore right? We can remove this now right?

File: Assets/Scripts/StaminaBar.cs [hunk 4]
```diff
@@ -70,4 +87,43 @@ void UpdateFillAmount(float newFill)
...
+    private IEnumerator FadeCoroutine(float startAlpha, float endAlpha, Image image)
+    {
+        float currentAlpha = startAlpha;
+        while (Mathf.Abs(currentAlpha - startAlpha) < Mathf.Abs(startAlpha - endAlpha))
+        {
+            image.color = new Color(image.color.r, image.color.g, image.color.b, currentAlpha);
+            currentAlpha += fadeSpeed * Time.deltaTime * Mathf.Sign(endAlpha - startAlpha);
+            yield return null;
+        }
+
```
Comment (jurcaua) [inline Assets/Scripts/StaminaBar.cs:116]: Force at the end to reach the endAlpha or else it will sometimes not go all the way:
```suggestion

image.color = new Color(image.color.r, image.color.g, image.color.b, endAlpha);
```

image

**Review by jurcaua — CHANGES_REQUESTED** [review]
Also remember to update the StaminaBar prefab and save all the changes on the prefab!

You may need to update the prefab to be an empty game object, and then have the Glow -> Black Outline -> etc. be the order in the object.
You save changes to the prefab like:
image

To fix the popping issue I would turn off Loop Time in the Glow animation:
image

And also make sure the first and last keyframes have the same alpha value (or else it will instantly change between them). Currently the start is 1 and end is 0.

File: Assets/Scripts/StaminaBar.cs (hunk 3 above)
Comment (justineechen) [reply Assets/Scripts/StaminaBar.cs:12]: Yes! took out glowoutlineimage

**Comment by justineechen** [comment]
When I go in my Glow animation it seems like the first key frame is 1 and the last is 1 too, im holding down the slider thing and putting it to the beginning and end and looking at the .a value so I'm not seeing that unless I'm looking at the wrong place?

**Comment by jurcaua** [comment]
Animation looks good! Once the prefab stuff is all done, it should be ready to merge into master! Make sure to set this up on the main scene as well and test it out!!

**Review by jurcaua — APPROVED** [review]
GOOD TO GO
